clouddriver-kubernetes/src/test/java/com/netflix/spinnaker/clouddriver/kubernetes/caching/view/provider/KubernetesDataProviderIntegrationTest.java

```diff
@@ -0,0 +1,1117 @@
+    softly.assertThat(frontendApplication).isNotNull();
+    if (frontendApplication != null) {
+      assertFrontendApplication(softly, frontendApplication);
+    }
+
+    KubernetesV2Application backendApplication = applicationLookup.get("backendapp");
+    softly.assertThat(frontendApplication).isNotNull();
+    if (frontendApplication != null) {
+      assertBackendApplication(softly, backendApplication);
+    }
+  }
+
+  @Test
+  void getApplicationsExpanded(SoftAssertions softly) {
+    // This is the same as the unexpanded test, as it seems like we ignore the flag.
+    Set<? extends Application> result = applicationProvider.getApplications(true);
+    softly.assertThat(result).hasSize(2);
+
+    Map<String, KubernetesV2Application> applicationLookup =
+        result.stream()
+            .collect(toImmutableMap(Application::getName, a -> (KubernetesV2Application) a));
+
+    KubernetesV2Application frontendApplication = applicationLookup.get("frontendapp");
+    softly.assertThat(frontendApplication).isNotNull();
+    if (frontendApplication != null) {
+      assertFrontendApplication(softly, frontendApplication);
+    }
+
+    KubernetesV2Application backendApplication = applicationLookup.get("backendapp");
+    softly.assertThat(frontendApplication).isNotNull();
+    if (frontendApplication != null) {
+      assertBackendApplication(softly, backendApplication);
+    }
+  }
+
+  @Test
+  void getApplication(SoftAssertions softly) {
+    KubernetesV2Application result =
+        (KubernetesV2Application) applicationProvider.getApplication("backendapp");
+    assertThat(result).isNotNull();
+    assertBackendApplication(softly, result);
+  }
+
+  @Test
+  void getInstance(SoftAssertions softly) {
+    KubernetesV2Instance result =
+        instanceProvider.getInstance(ACCOUNT_NAME, "backend-ns", "pod backend-v015-vhglj");
+    assertThat(result).isNotNull();
+    assertBackendCurrentServerGroupInstance(softly, result);
+  }
+
+  @Test
+  void getServerGroupManagers(SoftAssertions softly) {
+    Set<KubernetesV2ServerGroupManager> results =
+        serverGroupManagerProvider.getServerGroupManagersByApplication("frontendapp");
+    assertThat(results).hasSize(1);
+    if (!results.isEmpty()) {
+      assertFrontEndServerGroupManager(softly, results.iterator().next());
+    }
+  }
+
+  @Test
+  void getApplicationLoadBalancers(SoftAssertions softly) {
+    Set<KubernetesV2LoadBalancer> results =
+        loadBalancerProvider.getApplicationLoadBalancers("frontendapp");
+    assertThat(results).hasSize(1);
+    assertFrontendLoadBalancer(softly, results.iterator().next());
+  }
+
+  @Test
+  void getLoadBalancersByName(SoftAssertions softly) {
+    List<Details> results =
+        loadBalancerProvider.byAccountAndRegionAndName(
+            ACCOUNT_NAME, "frontend-ns", "service frontend");
+    // TODO(ezimanyi): This is a bug; we are not properly looking up the load balancer, so always
+    // return null
```

Comment:
😭  Oh boy, maybe this is why we make a live call for a service by name/namespace [here](https://github.com/spinnaker/clouddriver/blob/master/clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/op/manifest/KubernetesDeployManifestOperation.java#L262).

Reply:
Luckily it's [easy to fix](https://github.com/spinnaker/clouddriver/blob/2b78b4693dccce18b87e7fea20dd32100099a2c1/clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/caching/view/provider/KubernetesV2LoadBalancerProvider.java#L91), I just didn't include that in this PR.